src/main/java/skibidi/Storage.java

```diff
@@ -0,0 +1,73 @@
+package skibidi;
+
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Scanner;
+
+import skibidi.task.AbstractTask;
+
+/**
+ * Handles writing tasks to disk and loading tasks from file.
+ */
+public final class Storage {
+    private String dataPath;
+
+    public Storage(String dataPath) {
+        this.dataPath = dataPath;
+    }
+
+    /**
+     * Save given task list to storage data path. Overrides any existing data on the file.
+     */
+    public void saveTasksToDisk(TaskList taskList) {
+        File directory = new File("data");
+        if (!directory.exists()) {
+            directory.mkdir();
+        }
+        File file = new File(dataPath);
+        try {
+            if (file.exists() || file.createNewFile()) {
+                FileWriter writer = new FileWriter(file, false);
+                for (int i = 1; i <= taskList.size(); i++) {
+                    AbstractTask task = taskList.getTask(i);
+                    writer.write(task.serialize() + '\n');
+                }
+                writer.close();
+            } else {
+                System.out.println("Failed to save changes to file.");
+            }
+        } catch (TaskList.TaskNotFoundException err) {
```

Comment:
can apply polymorphism.

Eg.
catch (Exception e) {System.out.println(e.getMessage())}

Reply:
Good call, a generic Exception handler would suffice for cases where special handling is not needed.